modules/runner/src/main/java/org/apache/ignite/internal/app/NodePropertiesImpl.java

```diff
@@ -105,4 +108,9 @@ public boolean colocationEnabled() {
     public CompletableFuture<Void> stopAsync(ComponentContext componentContext) {
         return nullCompletedFuture();
     }
+
+    @Override
+    public Map<String, String> nodeAttributes() {
+        return Map.of(COLOCATION_FEATURE_FLAG, Boolean.toString(colocationEnabled()));
```

Comment:
It's a bit misleading, that you are using the system property as a node attribute name. Can we at least have a different java constant for this?

Reply:
On the contrary, I believe it makes the solution more consistent.

Reply:
How does it make the solution more consistent? It's a constant that is located in IgniteSystemProperties (which is for system properties), and here it is a key in the map. I'm ok with it having the same string value, but I think a separate constant would be more clear, because these two keys have a different semantical value.

Reply:
> How does it make the solution more consistent?

Because it's the projection of the corresponding IgniteSystemProperties constant or the value from the vault. It's literary node local value of colocation feature flag.

Reply:
As you can see, the colocation flag in the Vault uses a different constant